_overviews/scala3-book/taste-control-structures.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Control Structures
+type: section
+description: This section demonstrates Scala 3 control structures.
+num: 8
+previous-page: taste-vars-data-types
+next-page: taste-modeling
+---
+
+
+Scala has the programming language control structures you find in other languages, and also has powerful `for` expressions and `match` expressions:
+
+- `if`/`else`
+- `for` loops and expressions
+- `match` expressions
+- `while` loops
+- `try`/`catch`
+
+These structures are demonstrated in the following examples.
+
+
+
+## `if`/`else`
+
+Scala’s `if`/`else` control structure is similar to other languages:
+
+```scala
+if x < 0 then
+  println("negative")
+else if x == 0
+  println("zero")
+else
+  println("positive")
+```
+
+Note that this really is an _expression_ — not a _statement_. This means that it returns a value, so you can assign the result to a variable:
+
+```scala
+val x = if a < b then a else b
+```
+
+As you’ll see throughout this book, _all_ Scala control structures can be used as expressions.
```

Comment:
If am not mistaken, `while` loops are not expressions.

```suggestion
As you’ll see throughout this book, _all_ Scala control structures, except `while` loops, can be used as expressions.
```
or
```suggestion
As you’ll see throughout this book, _most_ Scala control structures can be used as expressions.
```

Reply:
While loops are indeed expressions, they just return unit: https://dotty.epfl.ch/docs/reference/syntax.html